lib/Hakyll/Web/Template/Context.hs

```diff
@@ -92,7 +94,7 @@ data ContextField
     = EmptyField
     | StringField String
     | forall a. ListField (Context a) [Item a]
-
+    | forall a. LexicalListField (forall b. Context b -> a -> Context b) [a]
```

Comment:
Lexical scoping was not possible with the current `ListField` constructor, needed something more advanced :-) Also requires RankNTypes, not sure whether that's a bad thing.
Maybe it's also possible to further generalise this, but I don't see a way to do it without breaking compatibility for everyone who directly uses `ListField`. Maybe these should be moved to a `Context.Internal` module anyway.